include/alpaka/core/ThreadPool.hpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2022 Antonio Di Pilato
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <condition_variable>
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <thread>
+#include <vector>
+
+namespace alpaka::core
+{
+    class ThreadPool
+    {
+        using taskType = std::packaged_task<void()>;
+
+    public:
+        ThreadPool() : num_threads(std::thread::hardware_concurrency())
+        {
+            for(int i = 0u; i < num_threads; ++i)
+                //! All the threads run the lambda function in loop
+                m_threads.emplace_back(std::thread(
+                    [this]
+                    {
+                        while(true)
+                        {
+                            taskType task;
```

Comment:
is it necessary to create a new `task` object for every iteration ?

Reply:
so I guess you suggest moving the variable atop the while loop?

Reply:
possibly - but I did not look into how heavy an object `std::packaged_task` is, and whether it is considered good practice to reuse them or not.

Reply:
It should be 64 bytes (as returned by `sizeof(std::packaged_task<void()>)`. I guess it would be better to reuse it. I tested it both in standalone and in alpaka (tests passed).